src/compiler/binder.ts

```diff
@@ -210,6 +216,24 @@ namespace ts {
             return propAccess.name.text;
         }
 
+        /**
+         * Returns the first invocation of an identifier 'require' with one argument,
+         * or undefined if that none exist.
+         */
+        function findCommonJSRequireCalls(node: Node): Node {

```

Comment:
so why not put this in the parser, so that SourceFile.externalModuleIndicator is sufficient all the time?